build-tools/packages/build-cli/src/library/commands/generateEntrypoints.ts

```diff
@@ -0,0 +1,413 @@
+			default: optionDefaults.outDir,
+			exists: true,
+		}),
+		outFilePrefix: Flags.string({
+			description: `File name prefix for emitting entrypoint declaration files. Pattern of '${unscopedPackageNameString}' within value will be replaced with the unscoped name of this package.`,
+			default: optionDefaults.outFilePrefix,
+		}),
+		outFileAlpha: Flags.string({
+			description: "Base file name for alpha entrypoint declaration files.",
+			default: optionDefaults.outFileAlpha,
+		}),
+		outFileBeta: Flags.string({
+			description: "Base file name for beta entrypoint declaration files.",
+			default: optionDefaults.outFileBeta,
+		}),
+		outFilePublic: Flags.string({
+			description: "Base file name for public entrypoint declaration files.",
+			default: optionDefaults.outFilePublic,
+		}),
+		outFileSuffix: Flags.string({
+			description:
+				"File name suffix including extension for emitting entrypoint declaration files.",
+			default: optionDefaults.outFileSuffix,
+		}),
+		node10TypeCompat: Flags.boolean({
+			description: `Optional generation of Node10 resolution compatible type entrypoints matching others.`,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const { mainEntrypoint, node10TypeCompat } = this.flags;
+
+		const packageJson = await readPackageJson();
+
+		const {
+			mapQueryPathToApiTagLevel,
+			mapApiTagLevelToOutput,
+			mapNode10CompatExportPathToData,
+		} = getOutputConfiguration(this.flags, packageJson, this.logger);
+
+		const promises: Promise<void>[] = [];
+
+		// Requested specific outputs that are not in the output map are explicitly
+		// removed for clean incremental build support.
+		for (const [outputPath, apiLevel] of mapQueryPathToApiTagLevel.entries()) {
+			if (
+				apiLevel !== undefined &&
+				typeof outputPath === "string" &&
+				!mapApiTagLevelToOutput.has(apiLevel)
+			) {
+				promises.push(fs.rm(outputPath, { force: true }));
+			}
+		}
+
+		if (node10TypeCompat && mapNode10CompatExportPathToData.size === 0) {
+			throw new Error(
+				'There are no API level "exports" requiring Node10 type compatibility generation.',
+			);
+		}
+
+		if (mapApiTagLevelToOutput.size === 0) {
+			throw new Error(
+				`There are no package exports matching requested output entrypoints:\n\t${[
+					...mapQueryPathToApiTagLevel.keys(),
+				].join("\n\t")}`,
+			);
+		}
+
+		promises.push(generateEntrypoints(mainEntrypoint, mapApiTagLevelToOutput, this.logger));
+
+		if (node10TypeCompat) {
+			promises.push(
+				generateNode10TypeEntrypoints(mapNode10CompatExportPathToData, this.logger),
+			);
+		}
+
+		await Promise.all(promises);
+	}
+}
+
+async function readPackageJson(): Promise<PackageJson> {
+	const packageJson = await fs.readFile("./package.json", { encoding: "utf8" });
+	return JSON.parse(packageJson) as PackageJson;
+}
+
+function getOutPathPrefix(
+	{ outDir, outFilePrefix }: { outDir: string; outFilePrefix: string },
```

Comment:
Nit: It isn't immediately clear to me what `outFilePrefix` is. Is it the file name without extension? Might be worth a comment.